tests/k8st/tests/test_bgp_advert.py

```diff
@@ -273,3 +325,68 @@ def check_routes_gone():
                 for cip in cluster_ips:
                     self.assertNotIn(cip, routes)
             retry_until_success(check_routes_gone, retries=10, wait_time=5)
+
+    def test_rr_filtering_svc_adv(self):
+        """
+        Test route with svc ip is not filtered on RR
+        """
+        with DiagsCollector():
+            # Create a network policy on node-A accepts traffic from the external node.
+            run("""docker exec -i kube-master kubectl apply -f - << EOF
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-tcp-80-ex
+  namespace: bgp-test
+spec:
+  podSelector: {node-A}
+  policyTypes:
+  - Ingress
+  ingress:
+  - from:
+    - ipBlock: { cidr: 20.192.0.5/32 }
+    ports:
+    - protocol: TCP
+      port: 80
+EOF
+""")
+           # Create RR host on node-B
+            cluster_id = str(IPAddress(0xFF000001))
+            rr = DockerHost('RR', start_calico=False)
+            rr.add_resource({
+                'apiVersion': 'projectcalico.org/v3',
+                'kind': 'Node',
+                'metadata': {
+                    'name': rr.get_hostname(),
+                    'labels': {
+                        'routeReflectorClusterID': cluster_id,
+                        },
+                    },
+                'spec': {
+                    'bgp': {
+                        'routeReflectorClusterID': cluster_id,
+                        },
+                    },
+                })
+            rr.start_calico_node()
+
+           # Disable node-to-node mesh
+            bgpconfig_as_num = 64514
+            peer_as_num = 64514
+            update_bgp_config(Node-A, nodeMesh=False, asNum=bgpconfig_as_num)
+            create_bgp_peer(Node-A, "global", rr.ip, peer_as_num)
```

Comment:
These calls won't Just Work for the k8st test, because calicoctl needs to be called in a different way (as above).

You could just make different versions of these utilities, that call calicoctl in the right way.  But I wonder if we should instead look at modifying the k8st setup so that these calls do just work as they are.